### PR TITLE
Update docs/examples/09-graph_communication.jl

### PR DESCRIPTION
**Pull request by fredrikekre**
This patch updates the graph communication example. In particular, it shows how to use `MPI.Dist_graph_neighbors_count` and `MPI.Dist_graph_neighbors!`.

### CODE REVIEW

**Comment by fredrikekre** [comment]
>  Build #1091 passed (54 hours, 47 minutes, 48 seconds)

Took some time! :D

**Comment by fredrikekre** [comment]
Bump!

File: docs/examples/09-graph_communication.jl [hunk 1]
```diff
@@ -41,85 +41,49 @@ end
 source = Cint[rank]
 graph_comm = MPI.Dist_graph_create(comm, source, degree, dest)
 
+# Query number of ranks that point to this rank, and number of ranks this rank point to
+indegree, outdegree, _ = MPI.Dist_graph_neighbors_count(graph_comm)
+
+# Query which ranks that point to this rank, and which ranks this rank point to
+inranks = Vector{Cint}(undef, indegree)
+outranks = Vector{Cint}(undef, outdegree)
+MPI.Dist_graph_neighbors!(graph_comm, inranks, outranks)
```
Comment (simonbyrne) [inline docs/examples/09-graph_communication.jl:50]: We need to make an allocating version of this function!

**Review by simonbyrne — APPROVED** [review]

File: docs/examples/09-graph_communication.jl (hunk 1 above)
Comment (fredrikekre) [reply docs/examples/09-graph_communication.jl:50]: You mean a function `Dist_graph_neighbors` that combine `Dist_graph_neighbors_count` and `Dist_graph_neighbors!`? Should it always return weights then, where weights are set to 1 for unweighted graphs? Or perhaps set to `nothing`?

Comment (simonbyrne) [reply docs/examples/09-graph_communication.jl:50]: I think returning `nothing` would make sense

**Pull request merged**